src/main/java/org/alfresco/trashcan/TrashcanCleaner.java

```diff
@@ -93,110 +94,87 @@ public TrashcanCleaner(NodeService nodeService, TransactionService transactionSe
     }
 
     /**
-     * 
+     *
+     * It splits and deletes the {@link java.util.List} of
+     * {@link org.alfresco.service.cmr.repository.NodeRef}s received as
+     * argument in batches of {@link #batchSize}.
+     *
+     * @param nodes
+     */
+    private void delete(List<NodeRef> nodes)
+    {
+        if (deleteBatchCount <= batchSize)
+        {
+            deleteNodes(nodes);
+            return;
+        }
+
+        partition(nodes, batchSize).forEach(this::deleteNodes);
+    }
+
+    /**
+     *
      * It deletes the {@link java.util.List List} of
      * {@link org.alfresco.service.cmr.repository.NodeRef NodeRef} received as
      * argument.
-     * 
+     *
      * @param nodes
      */
     private void deleteNodes(List<NodeRef> nodes)
     {
-        for (int i = nodes.size(); i > 0; i--) 
+        AuthenticationUtil.runAsSystem(() ->
         {
-            nodeService.deleteNode(nodes.get(i - 1));
-        }
+            RetryingTransactionCallback<Void> txnWork = () ->
+            {
+                nodes.forEach(nodeService::deleteNode);
```

Comment:
In the initial version of the code we used descending order to iterate over the nodes. Why do we need to change this?

Reply:
We don't need to, but it is a bit cleaner.
By testing both ways of iterating over the nodes, I found no importance in the order.